server/internals/service/service_test.go

```diff
@@ -210,3 +222,190 @@ func TestSaveX509Certificate_Success(t *testing.T) {
+	).Return(nil, fmt.Errorf("error"))
+	service := NewService(mockRepo)
+
+	_, err := service.GetZkUserMetadata(scopes, userID)
+
+	assert.NotNil(t, err)
+	assert.True(t, strings.HasPrefix(err.Error(), "failed to get country metadata:"))
+}
+
+func TestGetZkUserMetadata_FailedToGetEmailZkMetadata(t *testing.T) {
+	scopes := "country,email_verified"
+	mockRepo := &MockRepository{}
+
+	mockRepo.On(
+		"GetUserMetadata", userID, constants.USER_COUNTRY_METADATA_KEY,
+	).Return(&models.UserMetadata{UserID: userID}, nil)
+
+	mockRepo.On(
+		"GetUserMetadata", userID, constants.USER_EMAIL_VERIFIED_METADATA_KEY,
+	).Return(nil, fmt.Errorf("error"))
+
+	service := NewService(mockRepo)
+
+	_, err := service.GetZkUserMetadata(scopes, userID)
+
+	assert.NotNil(t, err)
+	assert.True(t, strings.HasPrefix(err.Error(), "failed to get email metadata:"))
+}
+
+func TestGetZkUserMetadata_FailedToGetDisplayNameMetadata(t *testing.T) {
+	scopes := "country,email_verified,display_name"
+	mockRepo := &MockRepository{}
+
+	mockRepo.On(
+		"GetUserMetadata", userID, constants.USER_COUNTRY_METADATA_KEY,
+	).Return(
+		&models.UserMetadata{
+			UserID: userID,
+			Key:    constants.USER_COUNTRY_METADATA_KEY,
+			Value:  country,
+		},
+		nil,
+	)
+
+	mockRepo.On(
+		"GetUserMetadata", userID, constants.USER_EMAIL_VERIFIED_METADATA_KEY,
+	).Return(
+		&models.UserMetadata{
+			UserID: userID,
+			Key:    constants.USER_EMAIL_VERIFIED_METADATA_KEY,
+			Value:  "true",
+		},
+		nil,
+	)
+
+	mockRepo.On(
+		"GetUserMetadata", userID, constants.USER_DISPLAY_NAME_METADATA_KEY,
+	).Return(nil, fmt.Errorf("error"))
+
+	service := NewService(mockRepo)
+
+	_, err := service.GetZkUserMetadata(scopes, userID)
+
+	assert.NotNil(t, err)
+	assert.True(t, strings.HasPrefix(err.Error(), "failed to get display name metadata:"))
+}
+
+func TestGetZkUserMetadata_Success(t *testing.T) {
+	scopes := "country,email_verified,display_name,color"
+	mockRepo := &MockRepository{}
+
+	mockRepo.On(
+		"GetUserMetadata", userID, constants.USER_COUNTRY_METADATA_KEY,
+	).Return(
+		&models.UserMetadata{
+			UserID: userID,
+			Key:    constants.USER_COUNTRY_METADATA_KEY,
+			Value:  country,
+		},
+		nil,
+	)
+
+	mockRepo.On(
+		"GetUserMetadata", userID, constants.USER_EMAIL_VERIFIED_METADATA_KEY,
+	).Return(
+		&models.UserMetadata{
+			UserID: userID,
+			Key:    constants.USER_EMAIL_VERIFIED_METADATA_KEY,
+			Value:  "true",
+		},
+		nil,
+	)
+
+	mockRepo.On(
+		"GetUserMetadata", userID, constants.USER_DISPLAY_NAME_METADATA_KEY,
+	).Return(
+		&models.UserMetadata{
+			UserID: userID,
+			Key:    constants.USER_DISPLAY_NAME_METADATA_KEY,
+			Value:  displayName,
+		},
+		nil,
+	)
+
+	service := NewService(mockRepo)
+
+	zkMetadata, err := service.GetZkUserMetadata(scopes, userID)
+
+	assert.Nil(t, err)
+	assert.Equal(t, country, zkMetadata.Country)
+	assert.Equal(t, displayName, zkMetadata.DisplayName)
+	assert.Equal(t, "red", zkMetadata.Color)
+	assert.True(t, zkMetadata.IsEmailVerified)
+}
```

Comment:
We should also add here unit tests for functions: `VerifyAuthorizationCode` and `GenerateAuthJwtCode`